openhands/runtime/base.py

```diff
@@ -213,6 +213,37 @@ async def on_event(self, event: Event) -> None:
             source = event.source if event.source else EventSource.AGENT
             self.event_stream.add_event(observation, source)  # type: ignore[arg-type]
 
+    def clone_repo(self, github_token: str | None, selected_repository: str | None):
+        if not github_token or not selected_repository:
+            return
+        url = f'https://{github_token}@github.com/{selected_repository}.git'
+        dir_name = selected_repository.split('/')[1]
+        action = CmdRunAction(
+            command=f'git clone {url} {dir_name} ; cd {dir_name} ; git checkout -b openhands-workspace'
+        )
+        logger.info(action, extra={'msg_type': 'ACTION'})
```

Comment:
Let's make this log message more descriptive instead of logging the action itself. E.g. `cloning repo {selected_repository}`

We should also use `self.log` here, which will include the runtime ID.

Ditto below